datahost-ld-openapi/src/tpximpact/datahost/ldapi/models/release_schema.clj

```diff
@@ -0,0 +1,73 @@
+(ns tpximpact.datahost.ldapi.models.release-schema
+  (:require
+   [malli.core :as m]
+   [malli.error :as me]
+   [tpximpact.datahost.ldapi.schemas.common :refer [registry]]
+   [tpximpact.datahost.ldapi.schemas.release-schema :as s.release-schema]
+   [tpximpact.datahost.ldapi.models.release :as models.release]
+   [tpximpact.datahost.ldapi.models.shared :as models.shared])
+  (:import [java.net URLEncoder]))
+
+(defn normalise-schema
+  [base-entity api-params jsonld-doc]
+  (when-not (s.release-schema/api-params-valid? api-params)
+    (throw (ex-info "Invalid API parameters"
+                    {:type :validation-error
+                     :validation-error (-> (m/explain s.release-schema/ApiParams
+                                                      api-params
+                                                      {:registry registry})
+                                           (me/humanize))})))
+  (let [{:keys [series-slug release-slug]} api-params]
+    (as-> jsonld-doc $
+      (models.shared/merge-params-with-doc api-params $)
+      (models.shared/validate-id release-slug $)
+      (models.shared/validate-context $ base-entity)
+      (assoc $ "@type" "dh:TableSchema"
+             ;;"@type" "dh:DimensionColumn"
+             "datahost:appliesToRelease" (format "/data/%s/releases/%s"
+                                                 (URLEncoder/encode series-slug "UTF-8") 
+                                                 (URLEncoder/encode release-slug "UTF-8"))
```

Comment:
If slugs are created correctly via `slugify or whatever we decide is correct at creation time, earlier in the stack before they hit this function, we shouldn't need to encode slugs here (?)

Reply:
That's true only when you can guarantee the arguments passed to `normalise-schema` came in via that 'slugify' code path. Which is hard to do long term (e.g. someone will eventually pass unencoded strings).